inspirehep/submissions/views.py

```diff
@@ -156,6 +165,158 @@ def start_workflow_for_submission(self, submission_data, control_number=None):
         abort(503)
 
 
+class JobSubmissionsResource(BaseSubmissionsResource):
+    decorators = [login_required_with_roles()]
+    user_allowed_status_changes = {
+        "pending": ["pending"],
+        "open": ["open", "closed"],
+        "closed": ["closed"],
+    }
+
+    def get(self, pid_value):
+        try:
+            pid, _ = pid_value.data
+            record = JobsRecord.get_record_by_pid_value(pid.pid_value)
+        except PIDDoesNotExistError:
+            abort(404)
+
+        serialized_record = job_v1.dump(record)
+        return jsonify({"data": serialized_record})
+
+    def post(self):
+        """Adds new job record"""
+        data = job_loader_v1()
+        data["status"] = "pending"
+        data = self.prepare_data(data)
+        record = JobsRecord.create(data)
+        db.session.commit()
+        self.create_ticket(record, "rt/new_job.html")
+        response = jsonify({"status": "created", "pid_value": record["control_number"]})
+        response.status_code = 201
+
+        return response
+
+    def put(self, pid_value):
+        """Updates existing record in db"""
+        data = job_loader_v1()
+        try:
+            pid, _ = pid_value.data
+            record = JobsRecord.get_record_by_pid_value(pid.pid_value)
+            if not self.user_can_edit(record):
+                abort(401)
+        except PIDDoesNotExistError:
+            abort(404)
+        data = self.prepare_data(data, record)
+        record.update(data)
+        db.session.commit()
+        self.create_ticket(record, "rt/update_job.html")
+        return jsonify({"status": "updated", "pid_value": record["control_number"]})
+
+    def prepare_new_record(self, data):
+        acquisition_source = self.get_acquisition_source()
+        if "$schema" not in data:
+            data["$schema"] = url_for(
+                "invenio_jsonschemas.get_schema",
+                schema_path="records/jobs.json",
+                _external=True,
+            )
+            data["status"] = "pending"
+
+        builder = JobBuilder(record=data)
+        if "acquisition_source" not in builder.record:
+            builder.add_acquisition_source(**acquisition_source)
+        return builder
+
+    def prepare_update_record(self, data, record):
+        # This contains all fields which can be removed from record (they are optional)
+        # if new value sent from the form is None, or empty in any other way
+        # (after de-serialization if it's missing from input data)
+        # this fields will be removed from record
+        additional_fields = [
+            "external_job_identifier",
+            "accelerator_experiments",
+            "urls",
+            "contact_details",
+            "reference_letters",
+        ]
+
+        if not is_superuser_or_cataloger_logged_in():
+            old_status = record.get("status", "pending")
+            new_status = data.get("status", old_status)
```

Comment:
shall we check if the status has changed?

Reply:
:+1: